src/client/components/2dContainer/blueprint.controller.js

```diff
@@ -20,38 +21,492 @@ export default class BlueprintEditor {
         this.$timeout = $timeout;
         this.boatParametersService = boatParametersService;
     }
-    /* I'll deal with this shit later
-    getBoundingSize(Maths, index0, index1) {
-        const copy = Maths;
-        let maxDiff = 0;
-        const vals = {
-            minVal: '',
-            maxVal: '',
+
+    // Acquire coordinates of frames
+    getFrameCoords(boat, lastY) {
+        let currY = lastY;
+        let i = 1;
+        let location1;
+        let location2;
+        let location3;
```

Comment:
I'm going to have a few comments wrt to these locations variables.. Firstly, don't declare variables until you need them. It's not apparent what the point of any of these variables do.